Live2D Android build issue

- axmol version: https://github.com/axmolengine/axmol/commit/05e3ddd1a1cec564d65cda826d1ff150f23fdbc3
- devices test on:
- developing environments
   - NDK version: r23c
   - Xcode version: 14.2
   - Visual Studio: 
      - VS version: 2019 (16.11), 2022 (17.6+)
      - MSVC version: 1929, 1934, 19.35, 19.36, 19.37
      - Windows SDK version: 10.0.22621.0
   - cmake version: 
Steps to Reproduce:

Live2d has the following in `Live2D/CMakeLists.txt`:
```
if (ANDROID)
    target_link_libraries(${LIB_NAME} INTERFACE Live2DCubismCore)
else()
    target_link_libraries(${LIB_NAME} Live2DCubismCore)
endif()
```

Then when it calls `setup_ax_extension_config(${LIB_NAME})`, it links with Axmol:
```
function(setup_ax_extension_config target_name)
  set(options DNTLINK opt_DNTLINK)
  cmake_parse_arguments(opt "" "${options}" ""
                          "" ${ARGN} )

  target_link_libraries(${target_name} ${_AX_CORE_LIB})
...
```

It results in the following error when running cmake for the Android build:
```
CMake Error at G:/game_engines/axmol/extensions/CMakeLists.txt:14 (target_link_libraries):
The keyword signature for target_link_libraries has already been used with
the target "Live2D".  All uses of target_link_libraries with a target must
be either all-keyword or all-plain.
```

Is the "INTERFACE" scope required for Android in this line?
`target_link_libraries(${LIB_NAME} INTERFACE Live2DCubismCore)`

If it is required, then a suggested fix:

Adjust `setup_ax_extension_config` to allow for optional link scope argument, for example: 

```
function(setup_ax_extension_config target_name)
  set(options DNTLINK opt_DNTLINK)
  set(options LINK_SCOPE opt_LINK_SCOPE)
  cmake_parse_arguments(opt "" "${options}" ""
                          "" ${ARGN} )

  target_link_libraries(${target_name} ${opt_LINK_SCOPE} ${_AX_CORE_LIB})
```

Which would be called like this:
`setup_ax_extension_config(${LIB_NAME} LINK_SCOPE PRIVATE)`

The above modification to `setup_ax_extension_config` has been tested, and it works, without affecting any other extensions.

--- a/extensions/CMakeLists.txt
+++ b/extensions/CMakeLists.txt
@@ -6,11 +6,12 @@ if(BUILD_SHARED_LIBS)
 endif()
 
 function(setup_ax_extension_config target_name)
-    set(options DNTLINK opt_DNTLINK)
-    cmake_parse_arguments(opt "" "${options}" ""
+  set(options DNTLINK opt_DNTLINK)
+  set(options LINK_SCOPE opt_LINK_SCOPE)
+  cmake_parse_arguments(opt "" "${options}" ""
                           "" ${ARGN} )
 
-    target_link_libraries(${target_name} ${_AX_CORE_LIB})
+  target_link_libraries(${target_name} ${opt_LINK_SCOPE} ${_AX_CORE_LIB})
 
   if(BUILD_SHARED_LIBS)
     target_compile_definitions(${target_name} PRIVATE _USREXDLL=1)
@@ -31,7 +32,7 @@ function(setup_ax_extension_config target_name)
   if(NOT opt_DNTLINK)
       set(_AX_EXTENSION_LIBS "${target_name};${_AX_EXTENSION_LIBS}" CACHE INTERNAL "extensions for auto link to target application")
   endif()
-  
+
   if(XCODE OR VS)
     ax_mark_code_files("${target_name}")
   endif()

--- a/extensions/Live2D/CMakeLists.txt
+++ b/extensions/Live2D/CMakeLists.txt
@@ -45,6 +45,11 @@ endif()
 ax_find_shaders("${CMAKE_CURRENT_LIST_DIR}/Framework/src/Rendering/axmol/shaders" LIVE2D_SHADER_SOURCES)
 ax_target_compile_shaders(${LIB_NAME} FILES ${LIVE2D_SHADER_SOURCES} CUSTOM)
 
-setup_ax_extension_config(${LIB_NAME})
+if (ANDROID)
+    ax_find_shaders("${CMAKE_CURRENT_LIST_DIR}/Framework/src/Rendering/axmol/shaders_tegra" LIVE2D_TEGRA_SHADER_SOURCES)
+    ax_target_compile_shaders(${LIB_NAME} FILES ${LIVE2D_TEGRA_SHADER_SOURCES} CUSTOM)
+endif()
+
+setup_ax_extension_config(${LIB_NAME} LINK_SCOPE PRIVATE)
 
 unset(LIB_NAME)

--- a/extensions/Live2D/Framework/src/Rendering/axmol/CubismRenderer_Cocos2dx.cpp
+++ b/extensions/Live2D/Framework/src/Rendering/axmol/CubismRenderer_Cocos2dx.cpp
@@ -747,8 +747,7 @@ void CubismShader_Cocos2dx::ReleaseShaderProgram()
 static const csmChar* VertShaderSrcSetupMask = "custom/live2d_setup_mask_vs";
 static const csmChar* FragShaderSrcSetupMask = "custom/live2d_setup_mask_fs";
 #if AX_TARGET_PLATFORM == AX_PLATFORM_ANDROID
-static const csmChar* FragShaderSrcSetupMaskTegra = "custom/live2d_setup_mask_fs";
-//"custom/live2d_setup_mask_tegra_fs";
+static const csmChar* FragShaderSrcSetupMaskTegra = "custom/live2d_setup_mask_tegra_fs";
 #endif
 
 
@@ -763,45 +762,38 @@ static const csmChar* VertShaderSrcMasked = "custom/live2d_masked_vs";
 // Normal & Add & Mult 共通
 static const csmChar* FragShaderSrc = "custom/live2d_fs";
 #if AX_TARGET_PLATFORM == AX_PLATFORM_ANDROID
-static const csmChar* FragShaderSrcTegra = "custom/live2d_fs";
-//"custom/live2d_tegra_fs";
+static const csmChar* FragShaderSrcTegra = "custom/live2d_tegra_fs";
 #endif
 
 // Normal & Add & Mult 共通 （PremultipliedAlpha）
 static const csmChar* FragShaderSrcPremultipliedAlpha = "custom/live2d_premultiplied_alpha_fs";
 #if AX_TARGET_PLATFORM == AX_PLATFORM_ANDROID
-static const csmChar* FragShaderSrcPremultipliedAlphaTegra = "custom/live2d_premultiplied_alpha_fs";
-//"custom/live2d_premultiplied_alpha_tegra_fs";
+static const csmChar* FragShaderSrcPremultipliedAlphaTegra = "custom/live2d_premultiplied_alpha_tegra_fs";
 #endif
 
 // Normal & Add & Mult 共通（クリッピングされたものの描画用）
 static const csmChar* FragShaderSrcMask = "custom/live2d_mask_fs";
 #if AX_TARGET_PLATFORM == AX_PLATFORM_ANDROID
-static const csmChar* FragShaderSrcMaskTegra = "custom/live2d_mask_fs";
-//"custom/live2d_mask_tegra_fs";
+static const csmChar* FragShaderSrcMaskTegra = "custom/live2d_mask_tegra_fs";
 #endif
 
 // Normal & Add & Mult 共通（クリッピングされて反転使用の描画用）
 static const csmChar* FragShaderSrcMaskInverted = "custom/live2d_mask_inverted_fs";
 #if AX_TARGET_PLATFORM == AX_PLATFORM_ANDROID
-static const csmChar* FragShaderSrcMaskInvertedTegra = "custom/live2d_mask_inverted_fs";
-//"custom/live2d_mask_inverted_tegra_fs";
+static const csmChar* FragShaderSrcMaskInvertedTegra = "custom/live2d_mask_inverted_tegra_fs";
 #endif
 
 // Normal & Add & Mult 共通（クリッピングされたものの描画用、PremultipliedAlphaの場合）
 static const csmChar* FragShaderSrcMaskPremultipliedAlpha = "custom/live2d_mask_premultiplied_alpha_fs";
 #if AX_TARGET_PLATFORM == AX_PLATFORM_ANDROID
-static const csmChar* FragShaderSrcMaskPremultipliedAlphaTegra = "custom/live2d_mask_premultiplied_alpha_fs";
-    //"custom/live2d_mask_premultiplied_alpha_tegra_fs";
+static const csmChar* FragShaderSrcMaskPremultipliedAlphaTegra = "custom/live2d_mask_premultiplied_alpha_tegra_fs";
 #endif
 
 // Normal & Add & Mult 共通（クリッピングされて反転使用の描画用、PremultipliedAlphaの場合）
 static const csmChar* FragShaderSrcMaskInvertedPremultipliedAlpha =
     "custom/live2d_mask_inverted_premultiplied_alpha_fs";
 #if AX_TARGET_PLATFORM == AX_PLATFORM_ANDROID
-static const csmChar* FragShaderSrcMaskInvertedPremultipliedAlphaTegra =
-    "custom/live2d_mask_inverted_premultiplied_alpha_fs";
-    //"custom/live2d_mask_inverted_premultiplied_alpha_tegra_fs";
+static const csmChar* FragShaderSrcMaskInvertedPremultipliedAlphaTegra = "custom/live2d_mask_inverted_premultiplied_alpha_tegra_fs";
 #endif
 
 CubismShader_Cocos2dx::CubismShader_Cocos2dx()

